AmplifyPlugins/Auth/Sources/AWSCognitoAuthPlugin/Service/ErrorMapping/AWSCongnitoIdentityProvider+AuthErrorConvertible.swift

```diff
@@ -0,0 +1,337 @@
+    var fallbackDescription: String { "Invalid parameter error" }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.invalidParameterError,
+            AWSCognitoAuthError.invalidParameter
+        )
+    }
+}
+
+extension InvalidPasswordException: AuthErrorConvertible {
+    var fallbackDescription: String { "Encountered invalid password." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.invalidPasswordError,
+            AWSCognitoAuthError.invalidPassword
+        )
+    }
+}
+
+extension LimitExceededException: AuthErrorConvertible {
+    var fallbackDescription: String { "Limit exceeded error." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.limitExceededError,
+            AWSCognitoAuthError.limitExceeded
+        )
+    }
+}
+
+extension NotAuthorizedException: AuthErrorConvertible {
+    var fallbackDescription: String { "Not authorized error." }
+
+    var authError: AuthError {
+        .notAuthorized(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.notAuthorizedError
+        )
+    }
+}
+
+extension PasswordResetRequiredException: AuthErrorConvertible {
+    var fallbackDescription: String { "Password reset required error." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.passwordResetRequired,
+            AWSCognitoAuthError.passwordResetRequired
+        )
+    }
+}
+
+extension ResourceNotFoundException: AuthErrorConvertible {
+    var fallbackDescription: String { "Resource not found error." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.resourceNotFoundError,
+            AWSCognitoAuthError.resourceNotFound
+        )
+    }
+}
+
+extension TooManyRequestsException: AuthErrorConvertible {
+    var fallbackDescription: String { "Too many requests error." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.tooManyRequestError,
+            AWSCognitoAuthError.requestLimitExceeded
+        )
+    }
+}
+
+extension UserNotConfirmedException: AuthErrorConvertible {
+    var fallbackDescription: String { "User not confirmed error." }
+
+    var authError: AuthError {
+        .service(
+            properties.message ?? fallbackDescription,
+            AuthPluginErrorConstants.userNotConfirmedError,
+            AWSCognitoAuthError.userNotConfirmed
+        )
+    }
+}
+
+extension UserNotFoundException: AuthErrorConvertible {
+    var fallbackDescription: String { "User not confirmed error." }
```

Comment:
Fallback description looks to be incorrect.

Reply:
Good catch, thanks! Addressed in [275c4c9](https://github.com/aws-amplify/amplify-swift/pull/3251/commits/275c4c948a8615a416f92300607dbc11d4735224).